### PR TITLE
Add notes edit find

### PR DESCRIPTION
**Pull request by Pinran-J**
Added:
- Find notes by keywords in title (No special characters, case-insensitive)
- Edit notes by title/content

### CODE REVIEW

**Review by ryanczx — APPROVED** [review]
Tested editNote and findNote, lgtm.

**Review by rui-han-crh — APPROVED** [review]

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -38,6 +38,38 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
...
+     *       </pre>
+     * @param sentence cannot be null
+     * @param word cannot be null, cannot be empty, must be a single word
+     */
+    public static boolean containsWordIgnoreCaseIgnoreSpecial(String sentence, String word) {
+        requireNonNull(sentence);
+        requireNonNull(word);
+
+        String preppedWord = word.trim();
+        System.out.println(preppedWord);
```
[the remark this answers is not in the document]
Comment (Pinran-J) [reply src/main/java/seedu/address/commons/util/StringUtil.java:60]: You are right, will remove it

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 2]
```diff
@@ -38,6 +38,38 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
...
+        checkArgument(!preppedWord.isEmpty(), "Word parameter cannot be empty");
+        checkArgument(preppedWord.split("\\s+").length == 1, "Word parameter should be a single word");
+
+        //String preppedSentence = sentence.replace("[^a-zA-Z]", " ");
```
[the remark this answers is not in the document]
Comment (Pinran-J) [reply src/main/java/seedu/address/commons/util/StringUtil.java:64]: You are right as well, will change

[the remark this answers is not in the document]
Comment (Pinran-J) [reply src/main/java/seedu/address/commons/util/StringUtil.java]: @ryanczx Fixed in [68ca6b8](https://github.com/AY2223S1-CS2103T-W12-2/tp/pull/81/commits/68ca6b895fcff3492c52701d97993c8f8a2b764c)
Please resolve after checking. Thank you!

File: src/main/java/seedu/address/commons/util/StringUtil.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pinran-J) [reply src/main/java/seedu/address/commons/util/StringUtil.java]: @ryanczx Fixed in [68ca6b8](https://github.com/AY2223S1-CS2103T-W12-2/tp/pull/81/commits/68ca6b895fcff3492c52701d97993c8f8a2b764c)
Please resolve after checking. Thank you!

**Pull request merged**